### PR TITLE
Admin and Teacher My Questions section shown fix.

### PR DESCRIPTION
**Pull request by realARCHaim**
https://youtrack.raccoongang.com/issue/AU-31

### CODE REVIEW

File: askup/utils/views.py [hunk 1]
```diff
@@ -616,3 +616,28 @@ def create_destroy_subscription(subject_id, user_id, subscribe):
...
+    :return: redirect to a correspondent section if it requires otherwise - None.
+    """
+    if organization_id and selected_organization is None:
+        # Case, when organization_id is specified in the link and restricted to this user
+        return redirect(reverse(url_name, kwargs={'user_id': user_id}))
+
+    if is_rank_list and selected_organization is None:
+        return redirect(reverse('askup:user_profile', kwargs={'user_id': user_id}))
+
+    return None
```
Comment (idegtiarov) [inline askup/utils/views.py]: None is default value and could be omitted

File: askup/views.py [hunk 2]
```diff
@@ -286,18 +298,25 @@ def user_profile_rank_list_view(request, user_id, organization_id=None):
...
-    if organization_id and selected_organization is None:
-        # Case, when organization_id is specified in the link and restricted to this user
-        return redirect(reverse('askup:user_profile_rank_list', kwargs={'user_id': profile_user.id}))
+    result_redirect = process_organization(
+        organization_id, selected_organization, profile_user.id, 'askup:user_profile_rank_list'
+    )
 
-    if selected_organization is None:
-        return redirect(reverse('askup:user_profile', kwargs={'user_id': profile_user.id}))
+    if result_redirect:
```
Comment (idegtiarov) [inline askup/views.py]: I expect it would be the check: `if result_redirect is not None:`

File: askup/views.py [hunk 3]
```diff
@@ -234,15 +236,25 @@ def get_queryset(self):
...
     )
 
-    if organization_id and selected_organization is None:
-        # Case, when organization_id is specified in the link and restricted to this user
-        return redirect(reverse('askup:user_profile', kwargs={'user_id': profile_user.id}))
+    result_redirect = process_organization(
+        organization_id, selected_organization, profile_user.id, 'askup:user_profile'
+    )
+
+    if result_redirect:
```
Comment (idegtiarov) [inline askup/views.py]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: askup/utils/views.py [hunk 4]
```diff
@@ -616,3 +616,28 @@ def create_destroy_subscription(subject_id, user_id, subscribe):
...
+    """
+    Process selected organization.
+
+    :return: redirect to a correspondent section if it requires otherwise - None.
+    """
+    if organization_id and selected_organization is None:
+        # Case, when organization_id is specified in the link and restricted to this user
+        return redirect(reverse(url_name, kwargs={'user_id': user_id}))
+
+    if is_rank_list and selected_organization is None:
```
Comment (idegtiarov) [inline askup/utils/views.py:631]: does it correct that in `case organization_id and selected_organization is None` we do not care about the `is_rank_list` parameter?

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: askup/utils/views.py (hunk 1 above)
Comment (realARCHaim) [reply askup/utils/views.py]: done, thanks

File: askup/views.py (hunk 2 above)
Comment (realARCHaim) [reply askup/views.py]: Good point. Thanks

File: askup/views.py (hunk 3 above)
Comment (realARCHaim) [reply askup/views.py]: done

File: askup/utils/views.py (hunk 4 above)
Comment (realARCHaim) [reply askup/utils/views.py:631]: that's right. This part of the function code is executed in both cases - in "user profile rank list" view and in "user profile" view.

File: askup/templates/askup/base.html [hunk 5]
```diff
@@ -46,11 +46,17 @@
                 <ul class="nav navbar-nav navbar-right"> 
                     {% if request.user.is_authenticated %}
                     <li>
-                        {% with id=request.user.id user=request.user %}
-                        <a class="float" href="{% url 'askup:user_profile' user.id %}">{{ user.email }}
-                            <span class="badge">{{ user.username }}</span>
-                        </a>
-                        {% endwith %}
+                        {% if is_admin or is_teacher %}
+                            {% with id=request.user.id user=request.user %}
```
Comment (elliottyates) [inline askup/templates/askup/base.html]: Looks like `id=request.user.id` could just be omitted here?

Comment (realARCHaim) [reply askup/templates/askup/base.html]: that's right. thanks. removing

File: askup/tests/test_functional/test_general.py [hunk 6]
```diff
@@ -1583,21 +1575,15 @@ class TestSubscriptionsMailing(TestCase):
...
-
     @client_user('student01', 'student01')
     @patch('askup.utils.general.do_send_subscriptions_to_recipients')
     def test_subscriptions_sending_no_actual_subscriptions(self, mock_do_send_subscriptions_to_recipients):
         """
         Test subscriptions sending.
         """
-        send_subscription_emails()
         mock_do_send_subscriptions_to_recipients.return_value = None
         mock_do_send_subscriptions_to_recipients.assert_not_called()
```
Comment (elliottyates) [inline askup/tests/test_functional/test_general.py:1585]: The assert will have to come after the method under test, to ensure we're actually testing the functionality stated.

File: askup/utils/views.py [hunk 7]
```diff
@@ -39,9 +39,9 @@
 QSET_QUESTION_FILTERS = {
     'all': ('All', 'All the questions'),
     'mine': ('Mine', 'My question'),
-    'others': ('Others', 'The questions of the others'),
-    'unanswered': ('Unanswered', 'The questions that I didn\'t answer before'),
-    'incorrect': ('Incorrect', 'The questions that I was incorrectly answered last time'),
+    'others': ('Others', 'Other\'s questions'),
```
Comment (elliottyates) [inline askup/utils/views.py]: I think this should be `'Others\' questions'` actually (apostrophe after the plural 's').

File: askup/tests/test_functional/test_general.py (hunk 6 above)
Comment (realARCHaim) [reply askup/tests/test_functional/test_general.py:1585]: That's right. But I've tested it 10 minutes ago and it seems like mock asserts are processing before the test method run. I've placed `send_subscription_emails()` in the top and in the bottom of the method. Changed it's assertion condition to `called` and `not_called` - it's working in both positions. :)
That's why in beforehand I didn't notice the order of those lines.

File: askup/utils/views.py (hunk 7 above)
Comment (realARCHaim) [reply askup/utils/views.py]: Got it. Thanks.

**Review by idegtiarov — APPROVED** [review]

File: askup/tests/test_functional/test_general.py (hunk 6 above)
Comment (elliottyates) [reply askup/tests/test_functional/test_general.py:1585]: @KhaimovMR `.assert_called()` does not work if it comes after the `send_subscription_emails()` call, which is good -- that's expected. So the test really needs to read:

```
send_subscription_emails()
mock_do_send_subscriptions_to_recipients.assert_not_called()
```

(the extra `return_value` step is irrelevant and can be omitted, both in this test case and the one below, `test_subscriptions_sending_with_subscriptions`)

**Pull request merged**